pkg/service/server.go

```diff
@@ -72,13 +74,17 @@ func (mgr *Manager) Start() {
 	go metrics.Start()
 
 	// restore in-progress events if crashed
-	inProgressEvents, err := getNodesByAnnotationKey(kube, InProgressAnnotationKey)
+	inProgressEvents, err := getNodesByAnnotationKeys(kube, InProgressAnnotationKey, QueueNameAnnotationKey)
 	if err != nil {
 		log.Errorf("failed to resume in progress events: %v", err)
 	}
 
 	// messages from in-progress are loaded to stream first
-	for node, sqsMessage := range inProgressEvents {
+	for node, annotations := range inProgressEvents {
+		if annotations[QueueNameAnnotationKey] != ctx.QueueName {
+			continue
```

Comment:
should we still continue if queue is empty? going back to the upgrade scenario mentioned above, we might want to still evaluate InProgressAnnotationKey if queue name is not annotated